Content.Server/Power/EntitySystems/BatterySystem.cs

```diff
@@ -103,14 +90,27 @@ private void OnEmpPulse(EntityUid uid, BatteryComponent component, ref EmpPulseE
             UseCharge(uid, args.EnergyConsumption, component);
         }
 
+        private void DirtyWithRateLimit(EntityUid uid, BatteryComponent battery)
+        {
+            // Limit network state update rate
+            if (_timing.CurTime > battery.NextSyncTime)
+            {
+                Dirty(uid, battery);
+                battery.NextSyncTime = _timing.CurTime + _syncDelay;
+            }
+        }
+
         public float UseCharge(EntityUid uid, float value, BatteryComponent? battery = null)
         {
-            if (value <= 0 ||  !Resolve(uid, ref battery) || battery.CurrentCharge == 0)
+            if (value <= 0 || !Resolve(uid, ref battery) || battery.CurrentCharge == 0)
                 return 0;
 
             var newValue = Math.Clamp(0, battery.CurrentCharge - value, battery.MaxCharge);
```

Comment:
While you're here, maybe fix the semantic :
```suggestion
            var newValue = Math.Clamp(battery.CurrentCharge - value, 0, battery.MaxCharge);
```
It has been here for a while, without any notable side-effects (it's working as-is).

Reply:
Huh, good catch!